### PR TITLE
Improve linting

### PR DESCRIPTION
**Pull request by Hypnosphi**
## What I did
— Fixed existing lint warnings
— Switched `remark` to quite mode to remove all the "this file is OK" noise
— Added `--report-unused-disable-directive` option to `lint:js` script, so that we don't use more disablings than we actually need
— Optimised eslint config: removed rules that duplicate airbnb config, fully ignored rules conflicting with Prettier
— Stopped ignoring accessibility rules and fixed all the violations
— Included `.storybook` and other previously ignored paths into linting

## How to test
```
$ yarn lint
yarn run v1.3.2
$ yarn lint:js . && yarn lint:ts **/*.ts && yarn lint:md .
$ cross-env NODE_ENV=production eslint --cache --cache-location=.cache/eslint --ext .js,.jsx,.json --report-unused-disable-directives .
$ tslint -p . -c tslint.json -t stylish '**/*.ts'
$ remark -q .
✨  Done in 10.34s.
```

### CODE REVIEW

File: examples/cra-kitchen-sink/src/stories/index.js [hunk 1]
```diff
@@ -90,7 +90,7 @@
 //   .add('with some emoji', () => (
```
Comment (igor-dv) [inline examples/cra-kitchen-sink/src/stories/index.js]: what is this file needed for ?

**Review by igor-dv — APPROVED** [review]

File: examples/cra-kitchen-sink/src/stories/index.js (hunk 1 above)
Comment (Hypnosphi) [reply examples/cra-kitchen-sink/src/stories/index.js]: Looks like @danielduan forgot to remove it
